konveyor/pelorus
# Image sha fix

## Describe the behavior changes introduced in this PR
This change builds upon PR https://github.com/redhat-cop/pelorus/pull/229 and introduces 2 new changes:

1. Modify the commit time exporter to look for the `build.spec.output.to.kind` and if it equals `DockerImage` then take the commit_sha instead of the deploy_sha to match against.
2.  Modify the deploy time exporter to allow regular expression matching against a commit sha without the prefix `sha256:`.

## Linked Issues?

resolves #222 


## Testing Instructions

Create a BuildConfig where the `spec.output.to.kind` is set to `DockerImage`.  Go through the normal build & deploy process to see the linkage created.  If you need an example to pull from I can supply one generically from my customer.

@redhat-cop/mdt
